front/pages/api/poke/workspaces/index.ts

```diff
@@ -189,44 +250,80 @@ async function handler(
             ],
           },
         ],
+        order: order,
       });
 
+      // if limit is above originalLimit, sort manually and then splice
+      if (limit > originalLimit) {
+        // Order by plan, entreprise first, then pro, then free and old free using isEntreprisePlan, isProPlan and isFreePlan, isOldFreePlan methods
+        workspaces.sort((a, b) => {
+          const planAPriority = getPlanPriority(a.subscriptions[0].plan.code);
+          const planBPriority = getPlanPriority(b.subscriptions[0].plan.code);
+
+          return planAPriority - planBPriority;
+        });
+
+        workspaces.splice(originalLimit);
+      }
+
       return res.status(200).json({
-        workspaces: workspaces
-          .map((ws): LightWorkpaceWithSubscriptionType => {
-            let subscription: SubscriptionType | null = null;
-            if (ws.subscriptions && ws.subscriptions.length > 0) {
-              subscription = renderSubscriptionFromModels({
+        workspaces: await Promise.all(
+          workspaces.map(async (ws): Promise<PokeWorkspaceType> => {
+            const subscription: SubscriptionType = renderSubscriptionFromModels(
+              {
                 plan: ws.subscriptions[0].plan,
                 activeSubscription: ws.subscriptions[0],
-              });
-            }
+              }
+            );
 
-            return {
+            const lightWorkspace: LightWorkspaceType = {
               id: ws.id,
               sId: ws.sId,
               name: ws.name,
               role: "admin" as const, // Explicitly type this as "admin"
               segmentation: ws.segmentation,
               whiteListedProviders: ws.whiteListedProviders,
               defaultEmbeddingProvider: ws.defaultEmbeddingProvider,
+            };
+
+            const auth = await Authenticator.internalAdminForWorkspace(ws.sId);
```

Comment:
We usually avoid admin role on a workspace and use builder role: `internalBuilderForWorkspace`.